apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/CustomScalarAdapters.kt

```diff
@@ -53,6 +53,18 @@ private constructor(
     } as Adapter<T>
   }
 
+  fun variables(): Set<String> {
+    if (variables == null) {
+      return emptySet()
+    }
+
+    return variables.valueMap.entries.filter {
+      it.value == true
```

Comment:
Could be a bit tricky to understand, should the name for `variables()` be more descriptive? Maybe `trueVariableNames()` (not great...) `variablesThatAreTrue()` ?  Or maybe return the Map as-is and filter in `Operations.kt` (or even in `AdapterCommon`) instead?

Reply:
In the world were `CustomScalarAdapters` becomes `AdapterContext`, I don't want to expose all variables because only boolean ones are actually needed during parsing. 

I went with plain `variables` because it's also the terminology used in `evaluate`: https://github.com/apollographql/apollo-kotlin/blob/72630f31ad8082c285e3a50e207345776ff86d77/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/BooleanExpression.kt#L112

Agree it's not great but don't really have good alternatives. `variablesSet`? `definedVariables`?

Reply:
I'm ok with `variables` 👍

Nitpick: I think it can be simplified as 
```kotlin
    return variables.valueMap.filter {
      it.value == true
    }.keys
```